dali/python/nvidia/dali/external_source.py

```diff
@@ -74,13 +74,13 @@ def to_numpy(x):
         inputs = []
         checked = False
         for datum in data:
-            info = _b.CheckDLPackCapsule(datum)
-            if not info[0] and not checked:
+            (is_dlpack, is_gpu_data) = _b.CheckDLPackCapsule(datum)
+            if not is_dlpack and not checked:
                 _check_data_batch(data, batch_size, layout)
                 checked = True
             if isinstance(datum, (_tensors.TensorCPU, _tensors.TensorGPU)):
                 inp = type(datum)(datum, layout=layout) if layout is not None else datum
-            elif hasattr(datum, "__cuda_array_interface__") or (info[0] and info[1]):
+            elif hasattr(datum, "__cuda_array_interface__") or (is_dlpack and is_gpu_data):
```

Comment:
I suggest separating the code paths:
```suggestion
            elif is_dlpack:
              if is_gpu_data:
                inp = _tensors.TensorGPU(datum, layout or "")
              else:
                inp = _tensors.TensorCPU(datum, layout or "")
            elif hasattr(datum, "__cuda_array_interface__"):
```

Reply:
I agree with this one.